src/mcp/shared/session.py

```diff
@@ -33,6 +35,12 @@
 SendRequestT = TypeVar("SendRequestT", ClientRequest, ServerRequest)
 SendResultT = TypeVar("SendResultT", ClientResult, ServerResult)
 SendNotificationT = TypeVar("SendNotificationT", ClientNotification, ServerNotification)
+SendNotificationInternalT = TypeVar(
```

Comment:
This code is ugly appears to need some python typechecking sorcery to tidy up though